### ISSUE TITLE
setChime toggles without checking current status

### ISSUE THREAD

**Issue by barros001** [haywirecoder/homebridge-envisalink-ademco#43]
I have a couple of automations that turns the chime on/off at certain times. It turns on every time I disarm my alarm and it turns off at 8pm. I noticed a few times that if I disarm my alarm while chime is on, chime would get turned off. Same with the 8pm automation, if chime is off, the automation turns it on. This is likely because HomeKit is sending the command (on/off) regardless of the current status, which somewhat makes sense (to ensure accessory is set to desired state).

the `setChime` method will simply issue the command to toggle without verifying current state, causing the issue described above. I ended up wrapping my automations around an `if` block that checks the current status of the chime before turning it on/off.

I was planning on opening a PR but I got a little confused with the code so I decided to open an issue to get some clarity. `setChime` and `getChime` use the `this.envisakitCurrentStatus` variable to keep track of the current status, as a [**boolean**](https://github.com/barros001/homebridge-envisalink-ademco/blob/3a3c0bacf77377ad1c65f8b8a438c471d0b49a52/accessories/customAccessory.js#L101). But after further digging into the code I noticed that `this.envisakitCurrentStatus` also tracks the bypass status, as a [**string**](https://github.com/barros001/homebridge-envisalink-ademco/blob/3a3c0bacf77377ad1c65f8b8a438c471d0b49a52/accessories/customAccessory.js#L65C24-L65C24)? I'm not sure if there's a reason for this of if this was just overlooked.

**Comment by haywirecoder** [comment]
Hi,

Chime and Bypass objects are separate, while variables envisakitCurrentStatus stored the status of each object, they are different types.  The Chime envisakitCurrentStatus is boolean and both the variable and the home kit button is is updated in real-time by a virtual keypad.  
What I think you experiencing is a delay effect. For example, when you turn off the chime button the homekit setChime function is called, upon completion homekit set the value of the button to off. The problem is the virtual keypad hasn't got updated yet from the panel, so the button is reset to the value of the virtual key value (ON - which is out of date), and a few sec later it gets the virtual keypad gets the update from the panel and reset it again to the right value (OFF). 
I have seen this type of issue prior with other functions and went ahead apply the same fix here. You can download the updates in index.js and accessories/customAccessory.js for update, envisalink.js may also be needed since it has other changes but not related to this fix, Or you can wait for the next release.

**Comment by barros001** [comment]
> What I think you experiencing is a delay effect. For example, when you turn off the chime button the homekit setChime function is called, upon completion homekit set the value of the button to off

This is not really the problem, although I did notice that as well (I figure it was a delay thing and didn’t pay much attention). The issue here is setChime will blindly send the command to the panel. If HomeKit tells it to turn it ON twice, it will turn on and then turn off. That’s because it’s a toggle command. Try to following and you will see it:

* Create a new automation m the home app (any trigger, you will not save it anyways)
* Select the Chime accessory and set it to turn ON
* Click Test This Automation
* Click again (click a few times)

You will see that every time you click Test This Automation the chime will toggle on/off. That's because HomeKit is telling the plugin to turn it on but the plugin is toggling it without verifying current status.

I opened a PR (#44) that should fix the issue. Try the above and then try again with the PR applied.
